### PR TITLE
feat: add component Select.tsx Select.module.scss

### PR DESCRIPTION
**Pull request by Lelik7777**
task for adding component Select from  09/18/2023

### CODE REVIEW

File: mw-webapp/src/component/select/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  /**
+   * Options list (Array<OptionType>)
+   */
+  options: OptionType[];
+}
+interface OptionType {
+  id: string;
+  value: string;
+  text: string;
+}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:26]: Move before interface SelectProps

File: mw-webapp/src/component/select/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  value: string;
+  /**
+   * Select's name (string)
+   */
+  name: string;
+  /**
+   * Options list (Array<OptionType>)
+   */
+  options: OptionType[];
+}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:21]: Don't repeat in () type of properties. Let's use DRY principle

File: mw-webapp/src/component/select/Select.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   */
+  value: string;
+  /**
+   * Select's name (string)
+   */
+  name: string;
+  /**
+   * Options list (Array<OptionType>)
+   */
+  options: OptionType[];
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:20]: Delete ; at the end of comments

File: mw-webapp/src/component/select/Select.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+export const Select: React.FC<SelectProps> = (props: SelectProps) => {
+  const [selected, setSelected] = useState(props.value);
+  const handleOnChange = (e: React.ChangeEvent<HTMLSelectElement>) => setSelected(e.target.value);
+  return (
+    <label>
+      {props.label}
+      <select name={props.name}
+        defaultValue={selected}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:35]: move propert to other line

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.tsx (hunk 1 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx]: done

Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 2 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:27]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 4 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:37]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/select/Select.tsx:27]: I was mean that for example name: string (then  no need to repeat it in comment). Select's anme is enough. With options the same)

Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:27]: * Options list 
   */
options: OptionType[];
is it enough?

Comment (Howl404) [reply mw-webapp/src/component/select/Select.tsx:27]: Maybe "Array of Options list" ?

Comment (tritonJS826) [reply mw-webapp/src/component/select/Select.tsx:27]: Guys,  i don't like to duplicate types in ts-docs, word list or even letter `s` in the end of word is enough

File: mw-webapp/src/component/select/Select.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+   * Select's name (string)
+   */
+  name: string;
+  /**
+   * Options list (array of options)
+   */
+  options: OptionType[];
+}
+
+export const Select: React.FC<SelectProps> = (props: SelectProps) => {
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:29]: component is a public member

File: mw-webapp/src/component/select/Select.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+          <option key={option.id}
+            value={option.value}
+          >
+            {option.text}
+          </option>
+        ))}
+      </select>
+    </label>
+  );
+};
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:52]: lets use radix component as base

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

**Comment by tritonJS826** [comment]
BTW we added storybook to the project, could you add a story for select component (example - button)

File: mw-webapp/src/component/select/Select.tsx (hunk 2 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:27]: got it 🙂

File: mw-webapp/src/component/select/Select.module.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+.select {
+  font-size: inherit;
```
[the remark this answers is not in the document]
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.module.scss:2]: done

File: mw-webapp/src/component/select/Select.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import styles from "src/component/select/Select.module.scss";
+import {useState} from "react";
+
+interface OptionType {
+  id: string;
+  value: string;
+  text: string;
+}
+
+interface SelectProps {
```
[the remark this answers is not in the document]
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:10]: done

File: mw-webapp/src/component/select/Select.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import styles from "src/component/select/Select.module.scss";
+import {useState} from "react";
+
+interface OptionType {
```
[the remark this answers is not in the document]
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:4]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 5 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:29]: sorry, but i don`t know what you mean

**Comment by Lelik7777** [comment]
> BTW we added storybook to the project, could you add a story for select component (example - button)

**Pull request closed without merging**